### PR TITLE
numpyro.contrib.module: only collect mutables in nnx_module() when th…

### PR DESCRIPTION
**Pull request by esennesh**
…ey exist

Testing Done: avoided registration of empty mutables in nnx ConvNet

### CODE REVIEW

File: numpyro/contrib/module.py [hunk 1]
```diff
@@ -484,8 +484,6 @@ def nnx_module(name, nn_module):
 
     mutable_holder = None
     if eager_other_state_dict:
-        mutable_holder = numpyro_mutable(name + "$state")
-    if mutable_holder is None:
```
Comment (fehiepsi) [inline numpyro/contrib/module.py:488]: I guess it is better to move this `if` inside the above `if`?

Comment (esennesh) [reply numpyro/contrib/module.py:488]: My patch deletes the lower `if` since by construction, `mutable_holder is None` is always true here.  We don't want to create an empty mutable-state no matter what, since there are inference algorithms and other things that depend upon the nonexistence of mutables (which is how I found this in the first place).

Comment (fehiepsi) [reply numpyro/contrib/module.py:488]: That makes sense. Thanks for clarifying! Could you adjust the failing test to cover the change? I think without batchnorm, we no longer have mutable state in that test.

**Comment by esennesh** [comment]
Yeah, of course.  I’ve had some fires to put out the past few days, but over the weekend I’m gonna look at what’s up with the test.  Wouldn’t BatchNorm actually maintain mutable state throughout execution?  So it should have a non-None mutable.

File: numpyro/contrib/module.py (hunk 1 above)
Comment (esennesh) [reply numpyro/contrib/module.py:488]: All right!  Changed the test and in the no-batchnorm, no-dropout case, we no longer expect to see any `'nn$state'` in the keys found.
